specs/release-automation/flow-for-automation.puml

```diff
@@ -0,0 +1,65 @@
+@startuml
+title Release Automation Process
+
+start
+group RA-1: Trigerring the Process
+    :User passes link to excel dictionary on OSF to use for the release;
+    :User passes OSF personal access token;
+    :User triggers release process;
+end group
+
+group RA-2: Creating the Release Files
+    :Retrieve Excel Dictionary;
+    :Parse metadata from files sheet;
+    :Construct file names, ending in ".csv" or ".xlsx";
+    :Populate file with information from appropriate sheet;
+    :Add optional headers;
+    :Store file in appropriate directory based on its upload location;
+end group
+
+group RA-3: Deploying the files to GitHub
+    :Check if previous release exists on GitHub in the appropriate directory;
+
+    if (No previous release on GitHub) then (yes)
+        :Create new release branch and move files to the GitHub file storage directory;
+    else (no)
+        if (Previous release is newer) then (yes)
+            :Stop process and throw an error;
+            stop
+        else (no)
+            :Move old files to archive directory and move new files to appropriate directory;
+        endif
+    endif
+    :Create PR from release branch to dev branch;
+end group
+group RA-4: Deploying the files to OSF
```

Comment:
I think this should be moved into a different workflow file one whose start is "Release branch merged to dev".

Reply:
So we should have two workflows files, one for the two trigger events below:

1. User triggers the workflow using the GitHub action tab
2. User merges a release branch into main